FreeRTOS/Test/CMock/smp/config_assert/config_assert_utest.c

```diff
@@ -0,0 +1,249 @@
+                             char * file,
+                             int line,
+                             int cmock_num_calls )
+{
+    if( !x )
+    {
+        assertionFailed++;
+
+        if( shouldAbortOnAssertion == pdTRUE )
+        {
+            Throw( configASSERT_E );
+        }
+    }
+}
+
+static void validate_and_clear_assertions( void )
+{
+    TEST_ASSERT_EQUAL( 1, assertionFailed );
+    assertionFailed = 0;
+}
+
+void * pvPortMalloc( size_t xSize )
+{
+    return unity_malloc( xSize );
+}
+
+void vPortFree( void * pv )
+{
+    return unity_free( pv );
+}
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    vFakeAssert_StubWithCallback( vFakeAssertStub );
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+void port_release_task_lock_cb( int  num_calls)
+{
+    pxCurrentTCBs[ 0 ]->xTaskRunState = -1; /* taskTASK_YIELDING */
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/** 
+ * @brief This test ensures that the code asserts when the TCB's xTaskRunState
+ *        is not equal to taskTASK_YIELDING ( -2 )
+ * 
+ * <b>Coverage</b> 
+ * @code{c} 
+ * prvCheckForRunStateChange( pxTCB ); 
+ * 
+ * configASSERT( pxThisTCB->xTaskRunState == taskTASK_YIELDING );
+ * @endcode 
+ */
+void test_prvCheckForRunStateChange_asssert_runstate_ne_task_yield (void )
+{
+    xSchedulerRunning = pdTRUE;
+    uxSchedulerSuspended = 0U;
+
+    TCB_t currentTCB;
+
+    pxCurrentTCBs[ 0 ]                    =  &currentTCB;
+    pxCurrentTCBs[ 0 ]->uxCriticalNesting = 0;
+    pxCurrentTCBs[ 0 ]->xTaskRunState     = -2; /* taskTASK_YIELDING */
+
+    vFakePortReleaseTaskLock_AddCallback(port_release_task_lock_cb);
+    vFakePortDisableInterrupts_ExpectAndReturn( pdTRUE );
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortGetTaskLock_Expect();
+    vFakePortGetISRLock_Expect();
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortAssertIfISR_Expect();
+    vFakePortCheckIfInISR_ExpectAndReturn( pdFALSE );
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortGetCoreID_ExpectAndReturn( 0);
+    vFakePortReleaseISRLock_Expect();
+    vFakePortReleaseTaskLock_Expect();
+
+    EXPECT_ASSERT_BREAK( vTaskEnterCritical() );
+
+    validate_and_clear_assertions();
+
+}
+
+/** 
+ * @brief This test ensures that the code asserts when the TCB's critical
+ *        nesting count is less than or equal to zero
+ * 
+ * <b>Coverage</b> 
+ * @code{c} 
+ * prvYieldForTask( pxTCB ); 
+ * 
+ * configASSERT( portGET_CRITICAL_NESTING_COUNT() > 0U );
+ * @endcode 
+ *
+ * configUSE_PREEMPTION == 1
```

Comment:
Probably we don't need the config comment here since we will have different FreeRTOSConfigXXX.h file.